components/engine_rocks/src/compact.rs

```diff
@@ -92,14 +86,33 @@ impl CompactExt for RocksEngine {
             return Ok(());
         }
 
+        self.compact_files_cf(cf, &input_files, Some(output_level))
+    }
+
+    fn compact_files_cf(
+        &self,
+        cf: &str,
+        files: &[String],
+        output_level: Option<i32>,
+    ) -> Result<()> {
+        let db = self.as_inner();
+        let handle = util::get_cf_handle(db, cf)?;
+        let cf_opts = db.get_options_cf(handle);
+        let output_level = output_level.unwrap_or(cf_opts.get_num_levels() as i32 - 1);
+        let output_compression = cf_opts
+            .get_compression_per_level()
+            .get(output_level as usize)
+            .cloned()
+            .unwrap_or(DBCompressionType::No);
+        let output_file_size_limit = cf_opts.get_target_file_size_base() as usize;
+
+        let max_subcompactions = cmp::min(num_cpus::get(), 32);
```

Comment:
May it be too large?

Reply:
seems yes, don't know why it's set that large before, will check